### ISSUE TITLE
133031045 Inconsistencies in ~Relocate batch~ process (Staffordshire)

### ISSUE THREAD

**Issue by FreeUKGenIssues** [FreeUKGen/FreeUKRegProductIssues#1194]
Issue reported by **STSCoord** at 2019-03-20 17:04:05 UTC
Time: 2019-03-20T16:49:06+00:00
Session ID: 4c928b8d4894d4d189cf5c3b449e31f2
Problem Page URL: [/freereg1_csv_files/update_registers?church=5407e121eca9eb024826661b](/freereg1_csv_files/update_registers?church=5407e121eca9eb024826661b)
Previous Page URL: [https://www.freereg.org.uk/freereg1_csv_files/update_churches?place=53cd4985eca9ebee4f0095a3](https://www.freereg.org.uk/freereg1_csv_files/update_churches?place=53cd4985eca9ebee4f0095a3)
Reported Issue:
The 3 windows work in different ways when actioning/actioned.

Place: 2 blank lines before "Select Place" - why? Having selected the place, then only the place is shown - OK.

Church: 2 blank lines before "Select Church" - why? Having selected the church, then only the church is shown - OK.

Register Type: 2 blank lines before first register type - no "Select Register Type", or whatever - why? Having selected the register type, then the register type is only highlighted among all the other register types.

If the idea of the first two windows is to reduce the possibility of errors, then it probably works but then the third window messes up this clarity.

    

![Screenshot](https://www.freereg.org.uk)

**Comment by PatReynolds** [comment]
We now have example files to be investigated. could be the same issue as an entry in two places (Kirkby).

@edickens to link files here.

**Comment by edickens** [comment]
From Ali
The most recent instance was yesterday, the link on a contact action request led to Unspecified.

The link now leads to Other Transcript, so best put this on back burner, next time it happens I'll get back to you staight away.

GNKNAR.WRYTHOBA3.CSV.2562 

 the changed records do continue to come up as Unspecified in a search, until they are re-processed, eg amending an error online leads to the relevant file being shown as Other Transcript.

- I've downloaded the batch and all entries correctly have the register type of TR, and not empty for Unspecified.  E

**Comment by edickens** [comment]
Seems to be fixed.  See October 2020.

**Comment by stsccfr** [comment]
I don't think this is fixed, at least not in the sense of good design.  Here is a screenshot of the UI for relocating a batch:

<img width="1772" height="788" alt="Image" src="https://github.com/user-attachments/assets/528b8074-5536-4567-9008-0c91c8249b84" />

As the original poster pointed out, there is an unnecessary blank line at the top of each menu list, and the first menu has the "Select Place" as a menu choice, which makes no sense. Also see #561 which is a request to have the first two menus set by default to the current Place and Church, the reason being that the vast majority of relocations are simply to set the register type.

**Comment by Gev1990123** [comment]
Have created a pull request for this https://github.com/FreeUKGen/MyopicVicar/pull/2852

@Vino-S - can you merge the code for @stsccfr to test?

**Comment by Gev1990123** [comment]
This is in dev, and ready to be tested @stsccfr

**Comment by stsccfr** [comment]
The menus are not populating correctly, even after waiting.  Upon selection of a Country, the County list should be populated only with the counties in that country, and similarly for the other menus.  Also "England" is showing in both the Country and the County menus.

**Comment by stsccfr** [comment]
Tested on Dev and it is working now. Marking ready for deployment.
